test/test_testing.py

```diff
@@ -155,7 +157,18 @@ def test_can_connect_default(self):
         c.get_plan()  # shouldn't raise ConnectionError
 
     def test_can_connect_begin_with_initial_hooks(self):
-        harness = Harness(CharmBase, meta='''
+        pebble_ready_calls = collections.defaultdict(int)
+
+        class MyCharm(CharmBase):
+            def __init__(self, *args):
+                super().__init__(*args)
+                self.framework.observe(self.on.foo_pebble_ready, self._on_pebble_ready)
+                self.framework.observe(self.on.bar_pebble_ready, self._on_pebble_ready)
```

Comment:
You'd almost rather have a way to catch all possible pebble_ready events, so that you could assert that it isn't called for arbitrary names. Presumably self.on['foo'].pebble_ready would also work here. (at least that is the magic that makes on['db'].relation_changed work.)

Reply:
There's probably a way to dynamically do this and ensure no other pebble-ready events are called, but per voice discussion, it doesn't seem worth it -- leaving as is.